### ISSUE TITLE
Continueing training doesn't seem to work

### ISSUE THREAD

**Issue by kootenpv** [Conchylicultor/DeepQA#10]
Everytime I continue with training I see a "globalStep" updated, but looking in Tensorboard it shows the loss like starting from scratch! It just seems like it starts over again:

globStep: 5181
maxLength: 10
watsonMode: False
hiddenSize: 256
numLayers: 2
embeddingSize: 32

In the image below you can see I tried to continue training 2 times:

![tensorboard](https://cloud.githubusercontent.com/assets/2746227/19234922/e532db92-8eee-11e6-83bb-31affec9e772.png)

Any idea what is going wrong? I'm NOT passing the `--reset` flag.

**Comment by Conchylicultor** [comment]
Strange, can you give me the command you try for both launching and restoring the model ? I'll try to replicate the issue.

**Comment by kootenpv** [comment]
I use both to start and to restore:

    python3.5 main.py

Thanks for looking into it.

**Comment by Conchylicultor** [comment]
I don't really understand. I just did a quick test running 1200 steps and restoring the model and it seems to restore the training correctly. However, I have an higher variance than in your screenshot, so your run and mine seems different.

Before relaunching the training, do you have a `model.ckpt` in the `save/model` directory ? What's happens if you try to test a trained model. Does that generate random words or does the model is correctly restored ? On which version of TensorFlow are you ? Also just to be sure when you stop the run, make sure you press Ctrl+C only once and not exit while the model is being saved.

**Comment by kootenpv** [comment]
Ah my apologies, I use my own dataset. I will share it soon! I actually forked your project and allow the use of a twitter username, upon which it downloads all their tweets (only if they are a reply to someone else). Like that you can define QA pair for your favorite twitter person :-) So this is the reaosn it looks different (though I actually still just run it without changing any features).

In your case, could you maybe try to have it train for like 10 minutes and then try to see what happens when you restart it? I think by that time a difference in loss should already be visible. My colleague actually noticed the same thing happening to him. The reason I think something might be wrong is because you can still see that it loads this .ini parameters file (with higher globalStep).

**Comment by Conchylicultor** [comment]
Ok, I'll try training longer.

Otherwise, make sure you use the same dataset when you retrain the model. More specifically, the vocabulary size should be the same and the words ids identical. Otherwise, this will confuse the model and you'll observe a drop in accuracy like you have.

**Comment by kootenpv** [comment]
@Conchylicultor Thanks, I do indeed caching. In fact, I have it directly work with the pickle that's once generated. I find that format quite easy to make.

**Comment by Conchylicultor** [comment]
I'm sorry, I could not reproduce the issue. I tried to retrain some previous models that I had trained for 2 days and the training continue as it was left.

In addition to the previous things I said, there is an other thing you can check: when a model is restored, the terminal should warned you with this message.
```
WARNING: Restoring previous model from /home/.../DeepQA/save/model-test_retrain/model.ckpt
```

**Comment by kootenpv** [comment]
@Conchylicultor In my case it does say that it is continuing (Restoring previous model), you can see that also from the globStep. Only in the loss-graph it looks very suspicious, it looks like it starts from scratch. Could you check if the loss continues from there as well? Thanks!

**Comment by Conchylicultor** [comment]
That's what I did when I said the training continue as it left. I checked on TensorBoard the loss and don't see any discontinuity.
Does this problem always appears or is it random ?
Can you also check if you have this issue if you restart from a clean repository (reclone the program without applying any code modifications)

**Comment by kootenpv** [comment]
My apologies. 

A fresh clone solved it. After further investigation, it had to do with continuing training with different parameters! It stills hows up under the same loss graph. I hadn't noticed I had given a different maxLength. This resulted in getting a new pickle. Thanks for checking it out though.

And perhaps you have an idea how we (or mostly I) could prevent this; maybe somehow hashing the params for a unique ID so that a different loss graph would be shown?

**Comment by Conchylicultor** [comment]
Ok thanks. I'll close the issue. The way it's currently done is to restore the previous maxLength  at the beginning  before creating the graph and loading the dataset. Maybe you load the dataset before the parameters are restored..
https://github.com/Conchylicultor/DeepQA/blob/master/chatbot/chatbot.py#L448

Otherwise, use the modelTag flag to have different models with different parameters in separate folders.
